core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -688,10 +688,11 @@ class SocketServerTest extends JUnitSuite {
   @Test
   def testRequestMetricsAfterStop(): Unit = {
     server.stopProcessingRequests()
-
-    server.requestChannel.metrics(ApiKeys.PRODUCE.name).requestRate.mark()
+    val version = ApiKeys.PRODUCE.latestVersion
```

Comment:
Would it make sense to modify this test case to use two distinct versions so that we can verify that the metric is updated correctly per-version?